### ISSUE TITLE
Enhancement: doCall() optionally warns if any arguments are ignored.

### ISSUE THREAD

**Issue by bokov** [HenrikBengtsson/R.utils#100]
A seemingly successful invocation of `doCall()` could make it very hard to notice incorrect output when a user misspells an argument or sends an argument that they incorrectly believe is understood by the target function.

If `.ignoreUnusedArgs` is true, and if there are in fact arguments being ignored, it would be useful to issue a warning listing the ignored arguments. To avoid false positives, the warning should exclude arguments that won't be ignored due to being listed in `alwaysArgs`. The warnings could be toggled on and off through an additional argument to `doCall()` and to preserve current behaviour warnings would be off by default.

If this is something that sounds helpful, I have a PR tested and ready to submit, please let me know. Thanks.

**Comment by HenrikBengtsson** [comment]
Sounds like a reasonable feature.  Maybe one could move toward having `.onUnusedArgs=c("ignore", "warn", "error")` with backward compatibility where `.ignoreUnusedArgs=TRUE` would result in `.onUnusedArgs="ignore"`.  What do you think?

I welcome a PR on code, help docs, and package unit tests on this.

**Comment by bokov** [comment]
Yes, I like that even better. I'll update my patch. Is it okay if a PR has more than one commit?

Coming soon.

**Comment by HenrikBengtsson** [comment]
Great. Yes, multiple commits are ok in PRs.

**Comment by bokov** [comment]
#101 is ready for review, please let me know if there are further steps I need to take.
